driver-core/src/main/java/com/datastax/driver/core/policies/TokenAwarePolicy.java

```diff
@@ -94,26 +141,23 @@ public void init(Cluster cluster, Collection<Host> hosts) {
     }
 
     /**
-     * Return the HostDistance for the provided host.
-     *
-     * @param host the host of which to return the distance of.
-     * @return the HostDistance to {@code host} as returned by the wrapped policy.
+     * {@inheritDoc}
+     * <p/>
+     * This implementation always returns distances as reported by the wrapped policy.
      */
     @Override
     public HostDistance distance(Host host) {
         return childPolicy.distance(host);
     }
 
     /**
-     * Returns the hosts to use for a new query.
+     * {@inheritDoc}
      * <p/>
-     * The returned plan will first return replicas (whose {@code HostDistance}
-     * for the child policy is {@code LOCAL}) for the query if it can determine
-     * them (i.e. mainly if {@code statement.getRoutingKey()} is not {@code null}).
-     * Following what it will return the plan of the child policy.
-     *
-     * @param statement the query for which to build the plan.
-     * @return the new query plan.
+     * The returned plan will first return local replicas for the query (i.e.
+     * replicas whose {@link HostDistance distance} according to the child policy is {@code LOCAL}),
+     * if it can determine them (i.e. mainly if the statement's
+     * {@link Statement#getRoutingKey(ProtocolVersion, CodecRegistry)} routing key}
+     * is not {@code null}); following what it will return the rest of the child policy's original query plan.
```

Comment:
It would also be good to add that the replicas will be ordered by the configured `ReplicaOrdering`.